File: src/Interpreter/depthOfNode.java
package Interpreter;

/**
 * Depth of a node
 */
public class depthOfNode extends Node {
    private final int depth;

    /**
     * Intermediate nodes
     */
    depthOfNode(Node parent, String label, int depth) {
        super(label);
        this.depth = depth;
        if (parent != null) {
            parent.addChild(this);
        }
    }


    depthOfNode(Node parent, String label, String value, int depth) {
        super(label, value);
        this.depth = depth;
        if (parent != null) {
            parent.addChild(this);
        }
    }
    int getDepth() {
        return depth;
    }

    @Override
    depthOfNode getParent() {

        return (depthOfNode) super.getParent();
    }
}

File: reCreate/main/INTERPRETER/depthOfNode.py
from .Node import Node

class DepthOfNode(Node):
    def __init__(self, parent, label, depth,value=None):
        if not value:
            super().__init__(label)
            self.depth = depth
            if parent is not None:
                parent.addChild(self)
        else:
            super().__init__(label, value)
            self.depth = depth
            if parent is not None:
                parent.addChild(self)


    def getDepth(self):
        return self.depth

    def getParent(self):
        return super().getParent()
